helm-service/controller/deployment_handler.go

```diff
@@ -36,8 +38,8 @@ func NewDeploymentHandler(keptnHandler Handler, mesh mesh.Mesh, onboarder Onboar
 
 // HandleEvent handles deployment.triggered events by first changing the new configuration and
 // afterwards applying the configuration in the cluster
-func (h *DeploymentHandler) HandleEvent(ce cloudevents.Event) {
-
+func (h *DeploymentHandler) HandleEvent(ctx context.Context, ce cloudevents.Event) {
+	defer ctx.Value(GracefulShutdownKey).(*sync.WaitGroup).Done()
```

Comment:
let's add a `nil` check for `ctx.Value()` and for the cast to `*sync.WaitGroup`, i.e.

```
val := ctx.Value(GracefulShutdownKey)
if val == nil {
	return 
}
if wg, ok := val.(*sync.WaitGroup); ok {
	wg.Done()
}
```